includes/multi-currency/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -152,9 +152,14 @@ public function override_selected_currency( $return ) {
 			return $order ? $order->get_currency() : $return;
 		}
 
-		$switch_id = $this->get_subscription_switch_id_from_superglobal();
-		if ( $switch_id ) {
-			return get_post_meta( $switch_id, '_order_currency', true );
+		// Ensure the subscription for a $switch_id is fetched only once per page load,
+		// avoiding an infinite loop that can occur on the product page when `get_subscription()` is used.
+		static $currency_fetched_for_switch_id = null;
```

Comment:
Using a [`static` variable](https://www.php.net/manual/en/language.variables.scope.php#language.variables.scope.static) here to cache this value for subsequent calls to the same function in a single page load, preventing an infinite loop that occurs otherwise.